test_data/asm/vm_to_vm.asm

```diff
@@ -0,0 +1,45 @@
+machine Main {
+
+    VM vm;
+
+    reg pc[@pc];
+    reg X[<=];
+    reg Y[<=];
+    reg Z[<=];
+    reg A;
+
+    instr add X, Y -> Z = vm.add
+    instr sub X, Y -> Z = vm.sub
+    instr assert_eq X, Y { X = Y }
+
+    // this should be the identity function: x + 1 - 1 = x
+    function main {
+        A <=Z= add(1, 1);
+        A <=Z= sub(A, 1);
+        assert_eq A, 1;
+    }
+}
+
+machine VM {
+
+    reg pc[@pc];
+    reg X[<=];
+    reg Y[<=];
+    reg Z[<=];
+    reg A;
+    reg B;
+
+    instr add X, Y -> Z { X + Y = Z }
+
+    instr sub X, Y -> Z { X - Y = Z }
+
+    function add x: field, y: field -> field {
```

Comment:
do we really need the types already?

Reply:
Good call. So the types are not checked anywhere, but I wasn't sure how to express the signature without them. Any ideas?

Reply:
maybe same as instr?

My main point here is that if we don't need the types it simplifies the PR. There are quite a few new features already, it would be good to have only the essential ones.

Reply:
But for `instr` you have registers. I could change to `function add x, y -> _, _`? But it wouldn't change any of the code, because `field` is not checked anywhere.

Reply:
ok